packages/ui/src/components/charts/hour-chart.tsx

```diff
@@ -0,0 +1,164 @@
+        return props.data.reduce((acc, cur) => {
+            const index = acc.findIndex((item) => item.timestamp.getHours() === cur.timestamp.getHours());
+            if (index !== -1) {
+                const existing = acc[index];
+                existing.value = cur.value;
+                existing.sensorId = cur.sensorId;
+
+                if (cur.valueOut) {
+                    if (!existing.valueOut) {
+                        existing.valueOut = cur.valueOut;
+                        existing.sensorId = cur.sensorId;
+                    }
+                }
+                if (cur.valueCurrent) {
+                    if (!existing.valueCurrent) {
+                        existing.valueCurrent = cur.valueCurrent;
+                        existing.sensorId = cur.sensorId;
+                    }
+                }
+            }
+
+            return acc;
+        }, fillArray);
+    }, [fillArray, props.data]);
+
+    return (
+        <>
+            {hasOutValues || hasCurrentValues ? (
+                <div className="flex flex-row items-center justify-end gap-2">
+                    <ChartSwitchButton
+                        active={activeChart === "value"}
+                        chart="value"
+                        onClick={setActiveChart}
+                        label="Verbrauch"
+                    />
+                    {hasOutValues ? (
+                        <ChartSwitchButton
+                            active={activeChart === "valueOut"}
+                            chart="valueOut"
+                            onClick={setActiveChart}
+                            label="Einspeisung"
+                        />
+                    ) : null}
+                    {hasCurrentValues ? (
+                        <ChartSwitchButton
+                            active={activeChart === "valueCurrent"}
+                            chart="valueCurrent"
+                            onClick={setActiveChart}
+                            label="Leistung"
+                        />
+                    ) : null}
+                </div>
+            ) : null}
+            <ChartContainer config={chartConfig} className="max-h-96 min-h-52 w-full">
+                <BarChart
+                    data={processedData}
+                    accessibilityLayer
+                    margin={{
+                        top: 10,
+                        right: 10,
+                        left: 10,
+                        bottom: 10,
+                    }}
+                >
+                    <ChartLegend content={<ChartLegendContent />} />
+                    <ChartTooltip content={<ChartTooltipContent />} />
+                    <XAxis
+                        dataKey="timestamp"
+                        type="category"
+                        tickFormatter={(value) => tickFormatter(value)}
+                        tickLine={false}
+                        interval="equidistantPreserveStart"
```

Comment:
Bei dem BarChart könnte man vielleicht auch noch die Y-Achse beschriften

Reply:
In der Legende steht Verbauch (kWh). Das sollte ich denke ich ausreichen